{{cookiecutter.project_slug}}/README.md

```diff
@@ -297,6 +259,96 @@ PATH.reset()                      # str(PATH): 'Channel'
 ```
 
 
+
+### Downloader (utils.downloader.py)
+
+`downloader.py` has a `read` function that can read from both urls and file paths.
+To use:
+
+```
+from utils.downloader import read
+
+local_file_content = read('/path/to/local/file.pdf')            # Load local file
+web_content = read('https://example.com/page')                  # Load web page contents
+js_content = read('https://example.com/loadpage', loadjs=True)  # Load js before getting contents
+
+```
+
+ The `loadjs` option will load any scripts before reading the contents of the page,
+ which can be useful for web scraping.
+
+If you need to use a custom session, you can also use the `session` option. This can
+be useful for sites that require login information.
+
+
+
+### HTMLWriter
+
+The HTMLWriter is a tool for generating zip files to be uploaded to Kolibri Studio
+
+First, open an HTMLWriter context:
+
+```
+from utils.html import HTMLWriter
+with HTMLWriter() as zipper:
+    # Add your code here
+```
+
+To write the main file, you will need to use the `write_main_file` method
+
+```
+contents = "<html><head></head><body>Hello, World!</body></html>"
+zipper.write_main_file(contents)
+```
+
+You can also add other files (images, stylesheets, etc.) using `write_file`:
+```
+# Returns path to file "styles/style.css"
+css_path = zipper.write_file("style.css", "body{padding:30px}", directory="styles")
+contents = "...<link href='{}' rel='stylesheet'></link>...".format(css_path)
+```
+
+
+
+#### HTML parsing using BeautifulSoup
```

Comment:
The motivation for putting this waaay up in the README (right after downloader) is that this is likely the first step a soucheffist will need to do---get the index.thml of the source page and try to extract links, etc.